### PR TITLE
feat: add configure button for addons

### LINKED ISSUE #174: Addon Settings
We've been discussing addon settings and how they should be implemented.

There are 2 cases:
- an addon can have settings, yet also be installed without any settings
- an addon requires settings, and cannot be installed without configuring it

For these purpose, we will have 2 `behaviorHints` in the addon manifest:
```
manifest {
    behaviorHints: {
        configurable: true,
        configurationRequired: true
    }
}
```

When both `behaviorHints.configurable` and `behaviorHints.configurationRequired` are `true`, then the "Install" button of the addon should read "Configure" and serve as an external URL to the `/configure` path of the addon's URL. (presuming the addon manifest URL is `https://this.addon.url/manifest.json`, then this external URL should point to `https://this.addon.url/configure`)

When `behaviorHints.configurable` is `true` and `behaviorHints.configurationRequired` is falsey, then the "Install" button of the addon should stay the same, and a configure button (small, with a gear icon) that serves as an external URL to the `/configure` path should be visible on the left side of the "Install" button.

**Visual goal of this task in the Desktop app:**
Screen Shot 2020-04-28 at 15 39 13

### PR DESCRIPTION
**Pull request by tymmesyde**
Closes #174 

localhost_8080_(TV) (27)
image
image

### CODE REVIEW

**Comment by Iheuzio** [comment]
It is not working over on edge (Version 113.0.1774.42 (Official build) (64-bit))

After clicking the addon button there are errors

image

File: src/routes/Addons/Addon/Addon.js [hunk 1]
```diff
@@ -84,9 +94,24 @@ const Addon = ({ className, id, name, version, logo, description, types, install
                 }
             </div>
             <div className={styles['buttons-container']}>
-                <Button className={installed ? styles['uninstall-button-container'] : styles['install-button-container']} title={installed ? t('ADDON_UNINSTALL') : t('ADDON_INSTALL')} tabIndex={-1} onClick={toggleButtonOnClick}>
-                    <div className={styles['label']}>{installed ? t('ADDON_UNINSTALL') : t('ADDON_INSTALL')}</div>
-                </Button>
+                <div className={styles['action-buttons-container']}>
+                    {
+                        !behaviorHints.configurationRequired && behaviorHints.configurable ?
```
Comment (Iheuzio) [inline src/routes/Addons/Addon/Addon.js:99]: Site loads up until this point, error encountered with configurationRequired, it is not defined and throws a type error

image

Comment (Iheuzio) [reply src/routes/Addons/Addon/Addon.js:99]: Error for reference:
image

Comment (tymmesyde) [reply src/routes/Addons/Addon/Addon.js:99]: Did you do `npm install` after checkout ?

**Pull request merged**